In [5]:
import pandas as pd
df = pd.read_csv('weight-height.csv')
df.head()

Gender     Height      Weight
0   Male  73.847017  241.893563
1   Male  68.781904  162.310473
2   Male  74.110105  212.740856
3   Male  71.730978  220.042470
4   Male  69.881796  206.349801

In [16]:
x = df.drop('Weight', axis=1)
x.head()

Gender     Height
0       0  73.847017
1       0  68.781904
2       0  74.110105
3       0  71.730978
4       0  69.881796

In [18]:
y = df['Weight']
y.head()

0    241.893563
1    162.310473
2    212.740856
3    220.042470
4    206.349801
Name: Weight, dtype: float64

In [19]:
x.Gender = x['Gender'].replace(['Male', 'Female'], [0,1])
df1.head()

Gender     Height
0       0  73.847017
1       0  68.781904
2       0  74.110105
3       0  71.730978
4       0  69.881796

In [43]:
df2 = df.copy()
df3 = df.copy()
df3.head()

Gender     Height      Weight
0       0  73.847017  241.893563
1       0  68.781904  162.310473
2       0  74.110105  212.740856
3       0  71.730978  220.042470
4       0  69.881796  206.349801

In [21]:
x.tail()

Gender     Height
8550       1  60.483946
8551       1  63.423372
8552       1  65.584057
8553       1  67.429971
8554       1  60.921791

In [27]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.3, random_state=42)

In [28]:
from sklearn.linear_model import LinearRegression

In [29]:
li = LinearRegression()

In [30]:
li.fit(x_train,y_train)

LinearRegression()

In [31]:
y_pred = li.predict(x_test)

In [34]:
y_pred

array([142.44868114, 186.0517281 , 198.42283706, ..., 102.94080643,
       190.83713617, 143.34786753])

In [36]:
from sklearn.metrics import mean_squared_error, r2_score

In [37]:
mse = mean_squared_error(y_test, y_pred)

In [38]:
mse

96.83734437830608

In [39]:
r2 = r2_score(y_test, y_pred)

In [40]:
r2

0.905911242442266

# KNN Regressor

In [41]:
from sklearn.neighbors import KNeighborsRegressor
kn = KNeighborsRegressor()

In [44]:
df2.head()

Gender     Height      Weight
0       0  73.847017  241.893563
1       0  68.781904  162.310473
2       0  74.110105  212.740856
3       0  71.730978  220.042470
4       0  69.881796  206.349801

In [48]:
X = df2.drop('Weight', axis=1)
X.head()

Gender     Height
0       0  73.847017
1       0  68.781904
2       0  74.110105
3       0  71.730978
4       0  69.881796

In [49]:
Y = df2['Weight']
Y.head()

0    241.893563
1    162.310473
2    212.740856
3    220.042470
4    206.349801
Name: Weight, dtype: float64

In [50]:
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.3, random_state=42)

In [51]:
kn.fit(X_train,Y_train)

KNeighborsRegressor()

In [52]:
Y_pred = kn.predict(X_test) 

In [53]:
Y_pred

array([142.1445657 , 181.38276924, 187.07075266, ..., 101.79654426,
       192.43702356, 145.3632227 ])

In [54]:
mse = mean_squared_error(Y_test, Y_pred)

In [55]:
mse

121.33528273624482

In [56]:
r2 = r2_score(Y_test,Y_pred)

In [57]:
r2

0.8821086423439015

Linear Regression model has mse value 96.83 where Knn Regression model has mse value 121.33 which is not good model compared to linear regression model. On the other hand, Linear regression has 0.90 accuracy but the knn regression model has 0.88 accuracy rate. So compared both model we can say that linear regession model is better than knn regression model.